gitlab/base.py

```diff
@@ -107,7 +120,16 @@ def __getattr__(self, name: str) -> Any:
                 try:
                     return self.__dict__["_parent_attrs"][name]
                 except KeyError:
-                    raise AttributeError(name)
+                    message = name
+                    if self._list_created_object:
+                        message = (
+                            f"Unable to find attribute {name!r}. This object "
+                            f"({self.__class__}) was created via a list() call and "
+                            f"only a subset of the data may be present. To ensure all "
+                            f"data is present get the object using a get(object.id) "
+                            f"call. See {_URL_ATTRIBUTE_ERROR} for details."
+                        )
+                    raise AttributeError(message)
```

Comment:
I tried this locally and I think we have a chance to improve the re-raises here and the message.
1. use the standard object AttributeError message `f"{type(self).__name__!r} object has no attribute {name!r}"`
2. raise `from` to clearly chain exceptions (but this was already missing before, just a nice to have)
3. format this a bit so it stands out for the user (otherwise it's just a long string). Maybe you'll find a better way, just an idea below.

Something like (sorry github wont let me add suggestions on deleted lines):

```python
            except KeyError:
                try:
                    return self.__dict__["_parent_attrs"][name]
                except KeyError as e:
                    message = f"{type(self).__name__!r} object has no attribute {name!r}"
                    if self._list_created_object:
                        message = (
                            f"{message}\n\n"
                            f"({self.__class__!r}) was created via a list() call\n"
                            f"and only a subset of the data may be present. To ensure\n"
                            f"all data is present get the object using a get(object.id)\n"
                            f"call. For more details, see:\n\n{_URL_ATTRIBUTE_ERROR}"
                        )
                    raise AttributeError(message) from e
```

This gives me:
```
...
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/nejc/repos/python-gitlab/gitlab/base.py", line 121, in __getattr__
    return self.__dict__["_parent_attrs"][name]
KeyError: 'bla'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/nejc/repos/python-gitlab/gitlab/base.py", line 132, in __getattr__
    raise AttributeError(message) from e
AttributeError: 'Project' object has no attribute 'bla'

(<class 'gitlab.v4.objects.projects.Project'>) was created via a list() call
and only a subset of the data may be present. To ensure
all data is present get the object using a get(object.id)
call. For more details, see:

https://python-gitlab.readthedocs.io/en/stable/faq.html#attribute-error-list
>>> 
```

Reply:
Done. I did use the `textwrap` library for wrapping it to 70 characters.